src/grpc/stream_infer_handler.h

```diff
@@ -115,6 +115,23 @@ class ModelStreamInferHandler
   static void StateWriteResponse(InferHandler::State* state);
   bool Finish(State* state);
 
+  // Simple structure that carries the payload needed for
+  // response release callback.
+  struct StreamResponseReleasePayload {
+    State* state_;
+    std::vector<std::shared_ptr<const SharedMemoryManager::SharedMemoryInfo>>
+        ref_shm_regions_;
+
+    StreamResponseReleasePayload(
+        State* state,
+        std::vector<
+            std::shared_ptr<const SharedMemoryManager::SharedMemoryInfo>>&&
+            ref_shm_regions)
+        : state_(state), ref_shm_regions_(ref_shm_regions)
+    {
+    }
+  };
```

Comment:
This can be group with the one define in `infer_handler.h`? Looks the same to me except for the naming.